plugins/modules/content_upload.py

```diff
@@ -157,6 +171,11 @@ def main():
         elif module.foreman_params['repository']['content_type'] == 'file':
             query = 'name = "{0}" and checksum = "{1}"'.format(filename, checksum)
             content_unit = module.find_resource('file_units', query, params=repository_scope, failsafe=True)
+        elif module.foreman_params['repository']['content_type'] == 'ostree':
+            query = 'name = "{0}" and checksum = "{1}"'.format(filename, checksum)
+            ostree_type_scope = {'content_type': 'ostree_ref'}
```

Comment:
I would, for "safety" reasons, add the repo scope here. The search might work without, but if we set this, it will be limited to a specific repo, and should be faster.

```suggestion
            ostree_type_scope = {'content_type': 'ostree_ref'}
            ostree_type_scope.update(repository_scope)
```